src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -303,66 +362,239 @@ subroutine init_atm_static(mesh, dims, configs)
+               ! process' cells, if all the pixels of a completely to halo cells, then we know
+               ! we don't need to add this tiles neighbors to the stack.
+               ! 
+               ! However, if a single pixel maps to one of our cells then to be safe we should
+               ! push on its surrounding tiles.
+               if (res % cell <= nCellsSolve) then
+                  ALL_PIXELS_MAPPED_TO_HALO_CELLS = .FALSE.
+               endif
+
+            ! For outermost boundary cells, additional work is needed to verify that the pixel
+            ! actually lies within the nearest cell
+            else
+               zPixel = sphere_radius * sin(lat_pt)                  ! Model cell coordinates assume a "full" sphere radius
+               xPixel = sphere_radius * cos(lon_pt) * cos(lat_pt)    ! at this point, so we need to ues the same radius
+               yPixel = sphere_radius * sin(lon_pt) * cos(lat_pt)    ! for source pixel coordinates
+
+               if (in_cell(xPixel, yPixel, zPixel, xCell(res % cell), yCell(res % cell), zCell(res % cell), &
+                           nEdgesOnCell(res % cell), verticesOnCell(:,res % cell), xVertex, yVertex, zVertex)) then
+
+                  ter_integer(res % cell) = ter_integer(res % cell) + int(tval, kind=I8KIND)
+                  nhs(res % cell) = nhs(res % cell) + 1
+
+                  ! Similarly, if we have a border cell that is not a halo cell, then we should
+                  ! grab the current tiles surrounding tiles.
+                  if (res % cell <= nCellsSolve) then
+                     ALL_PIXELS_MAPPED_TO_HALO_CELLS = .FALSE.
+                  endif
+
+               endif
+            endif
+         enddo
+      enddo
+
+      if (ALL_PIXELS_MAPPED_TO_HALO_CELLS) then
+         ! Pass - All tiles mapped to its halo cells, so don't add any tiles to the stack
+      else
+         ! Calculate the tile neighbors of the current tile and push them onto the stack
+         call mpas_get_tile_neighbors(tile, up, down, left, right, topo_pool)
+         call push_tile_neighbors(stack, up, down, left, right, hash, topo_pool)
+      endif
+    enddo
+ enddo
+
+ ! Since we are taking an average of all the pixels that map to  a cell, divide each cell's 
+ ! terrain value by the number of pixels that we alloted to it.
+ do i = 1, nCells, 1
+    ter(i) = real(ter_integer(i), kind=R8KIND) / real(nhs(i), kind=R8KIND)
+ enddo
+
+ ! Use the hash table to deallocate each tile and then the hash itself.
+ do i = 0, nTileX, 1
+    do j = 0, nTileY, 1
+       if(associated(hash(i, j) % ptr)) then
+          deallocate(hash(i, j) % ptr % tile)
+          deallocate(hash(i, j) % ptr)
+       endif
+    enddo
+ enddo
+ deallocate(hash)
+
+ ! Deallocate the kdcords
+ do i = 1, nCells, 1 ! TODO: Update mpas_kd_tree so we can call that module to free this
+    deallocate(kdcords(i) % point)
+ enddo
+ deallocate(kdcords)
```

Comment:
Every static field will use the KD Tree, so lets move this down to the end of this function.